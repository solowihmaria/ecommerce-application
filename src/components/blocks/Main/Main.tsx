import styles from './Main.module.scss';
import { AboutSection } from './parts/About/About';
import { Hero } from './parts/Hero/Hero';
import { Testimonials } from './parts/Testimonials/Testimonials';
import { Newsletter } from './parts/Newsletter/Newsletter';

export const Main = () => (
    <main className={styles.main}>
        <Hero />
        <AboutSection />
        <Testimonials />
        <Newsletter />
    </main>
);
