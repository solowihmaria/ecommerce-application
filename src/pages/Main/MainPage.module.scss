.main {
    display: flex;
    justify-content: center;
    padding: 40px 20px;
}

.titleWrapper {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
}
