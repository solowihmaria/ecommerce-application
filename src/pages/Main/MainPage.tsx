import React from 'react';
import { Header } from '../../components/blocks/Header/Header';
import styles from './MainPage.module.scss';

export const MainPage = () => (
    <div data-testid="main-page-test-id">
        <Header />
        <main className={styles.main}>
            <div className={styles.titleWrapper}>
                <h1 className={styles.title}>Main Page Content</h1>
            </div>
        </main>
    </div>
);
