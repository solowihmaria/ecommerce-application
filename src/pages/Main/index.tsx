import React from 'react';
import { Link } from 'react-router-dom';

export const MainPage = (): React.JSX.Element => (
    <div>
        <h1>Главная</h1>
        <nav>
            <Link to="/login">Вход</Link> |{' '}
            <Link to="/register">Регистрация</Link>
        </nav>
    </div>
);
