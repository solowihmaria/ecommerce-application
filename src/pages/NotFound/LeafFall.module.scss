.leafContainer {
    position: absolute;
    z-index: 1;
    top: 0;
    left: 0;
    overflow: hidden;
    width: 100%;
    height: 100%;
    pointer-events: none;
}

.leaf {
    position: absolute;
    top: -200px;
    width: 150px;
    height: 150px;
    opacity: 0.8;
    background-image: url('../../assets/images/leaf.png');
    background-repeat: no-repeat;
    background-size: contain;
    animation: fall linear infinite;
}

@keyframes fall {
    to {
        transform: translateY(110vh) rotate(360deg);
        opacity: 0;
    }
}
