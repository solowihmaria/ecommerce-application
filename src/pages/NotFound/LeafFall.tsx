import { useEffect } from 'react';
import styles from './LeafFall.module.scss';

export const LeafFall = () => {
    useEffect(() => {
        const container = document.querySelector(`.${styles.leafContainer}`);

        for (let i = 0; i < 15; i++) {
            const leaf = document.createElement('div');
            leaf.className = styles.leaf;
            leaf.style.left = `${Math.random() * 100}%`;
            leaf.style.animationDuration = `${5 + Math.random() * 5}s`;
            leaf.style.animationDelay = `${Math.random() * 5}s`;

            container?.appendChild(leaf);
        }
    }, []);

    return <div className={styles.leafContainer}></div>;
};
