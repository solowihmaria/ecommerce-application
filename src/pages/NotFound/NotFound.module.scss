.notfound {
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100vh;
    padding: 2rem;
    text-align: center;
    background-color: #f8f9fa;
}

.content {
    max-width: 26rem;
    margin: auto;
    padding: 2rem;
}

.errorcode {
    margin-bottom: 0;
    font-size: 8rem;
    font-weight: 800;
    line-height: 1;
    color: #dc3545;
    text-shadow: 0 0.125rem 0.25rem rgb(0 0 0 / 10%);
}

.title {
    margin: 0.5rem 0;
    font-size: 2rem;
    font-weight: 700;
    color: #212529;
}

.message {
    margin: 1rem 0 1.5rem;
    font-size: 1.125rem;
    line-height: 1.5;
    color: #6c757d;
}

.homelink {
    display: inline-block;
    margin-top: 1rem;
    text-decoration: none;
}

.button {
    min-width: 12.5rem;
}

@media (width <= 768px) {
    .errorcode {
        font-size: 5rem;
    }

    .title {
        font-size: 1.75rem;
    }

    .content {
        padding: 1rem;
    }
}
