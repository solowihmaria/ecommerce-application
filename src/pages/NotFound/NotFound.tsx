import React from 'react';
import { Link } from 'react-router-dom';
import Heading from '../../components/ui/Heading/Heading';
import Button from '../../components/ui/Button/Button';
import styles from './NotFound.module.scss';

export const NotFoundPage: React.FC = () => {
    return (
        <div className={styles.notfound}>
            <div className={styles.content}>
                <div className={styles.errorcode}>404</div>
                <Heading level="h1" className={styles.title}>
                    Page Not Found
                </Heading>
                <p className={styles.message}>
                    Oops! The page you&apos;re looking for doesn&apos;t exist or
                    has been moved.
                </p>
                <Link to="/" className={styles.homelink}>
                    <Button className={styles.button}>Go to Homepage</Button>
                </Link>
            </div>
        </div>
    );
};
