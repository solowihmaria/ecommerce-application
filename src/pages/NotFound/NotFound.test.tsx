import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { NotFoundPage } from './NotFound';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { MainPage } from '../Main';

describe('<NotFoundPage/> component', () => {
    test('Render NotFoundPage component', () => {
        render(
            <MemoryRouter>
                <NotFoundPage />
            </MemoryRouter>
        );
        const notFoundPage = screen.getByTestId('test-id-not-found-page');
        expect(notFoundPage).toBeInTheDocument();
        expect(notFoundPage).toMatchSnapshot();
    });
    test('Render NotFoundPage component - click Go Home Button', async () => {
        render(
            <MemoryRouter initialEntries={['/test']}>
                <Routes>
                    <Route path="/" element={<MainPage />} />
                    <Route path="/test" element={<NotFoundPage />} />
                </Routes>
            </MemoryRouter>
        );
        const button = screen.getByRole('button', { name: /go to homepage/i });
        await userEvent.click(button);
        const mainPage = screen.getByTestId('main-page-test-id');
        expect(mainPage).toBeInTheDocument();
    });
});
