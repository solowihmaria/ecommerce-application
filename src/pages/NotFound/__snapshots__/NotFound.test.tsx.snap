// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage/> component Render NotFoundPage component 1`] = `
<div
  class="notFound"
  data-testid="test-id-not-found-page"
>
  <div
    class="content"
  >
    <div
      class="errorCode"
    >
      404
    </div>
    <h1
      class="heading heading--h1 title"
    >
      Page Not Found
    </h1>
    <p
      class="message"
    >
      Oops! The page you're looking for doesn't exist or has been moved.
    </p>
    <a
      class="homeLink"
      data-discover="true"
      href="/"
    >
      <button
        class="button primary button"
        type="button"
      >
        Go to Homepage
      </button>
    </a>
  </div>
</div>
`;
